src/Util/DateTimeFormat.php

```diff
@@ -184,17 +184,17 @@ public static function convert($s = 'now', $tz_to = 'UTC', $tz_from = 'UTC', $fo
 	 */
 	public static function fix(string $dateString): string
 	{
-		$patterns = [
-			['#(\w+), (\d+/\d+/\d+) - (\d+:\d+)#', '$1, $2 $3'],
-			['#(\d+-\d+-\d+)T(\d+:\d+:\d+)ZZ#', '$1T$2Z'],
-			['#(\d+-\d+-\d+)T(\d+:\d+:\d+\.\d+)ZZ#', '$1T$2Z'],
+		$search  = ['Mär', 'März', 'Mai', 'Juni', 'Juli', 'Okt', 'Dez', 'ET' , 'ZZ', ' - ', '&#x2B;'];
```

Comment:
We mustn't use these simple patterns to fix the problem. We are not only calling this function with fixed date values but also with date formulas. When we replace " - " with "" then a date formula like "now - 3 days" will stop working. I accidentally deleted many posts this way some weeks ago :grin:

Reply:
This is where function comments are helping in a big way. As long as this kind of knowledge stays in your head, it can only help yourself. As soon as you put it down for others, it helps everybody save their time.

Reply:
And adding each edge case a test so we can avoid such regressions ;-)))

Reply:
Problem is that it is not really an edge case. It is based on the way that functionality works. This functionality is called both with static values and with formulas. And thus problems like these can always occur.